owrx/source/afedri.py

```diff
@@ -0,0 +1,99 @@
+from owrx.source.soapy import SoapyConnectorSource, SoapyConnectorDeviceDescription
+from owrx.form.input import Input, CheckboxInput, NumberInput
+from owrx.form.input.device import TextInput
+from owrx.form.input.validator import RangeValidator
+from owrx.form.input.converter import OptionalConverter
+from owrx.form.input.validator import RequiredValidator
+from typing import List
+
+
+AFEDRI_DEVICE_KEYS = ["rx_mode"]
+AFEDRI_PROFILE_KEYS = ["r820t_lna_agc", "r820t_mixer_agc"]
+
+
+class AfedriAddressPortInput(TextInput):
+    def __init__(self):
+        super().__init__(
+            "afedri_adress_port",
+            "Afedri IP and Port",
+            infotext="Afedri IP and port to connect to. Format = IP:Port",
+            converter=OptionalConverter(),
+            validator=RequiredValidator(),
+        )
+
+
+class AfedriSource(SoapyConnectorSource):
+    def getSoapySettingsMappings(self):
+        mappings = super().getSoapySettingsMappings()
+        mappings.update({x: x for x in AFEDRI_PROFILE_KEYS})
+        return mappings
+
+    def getEventNames(self):
+        return super().getEventNames() + ["afedri_adress_port"] + AFEDRI_DEVICE_KEYS
+
+    def getDriver(self):
+        return "afedri"
+
+    def buildSoapyDeviceParameters(self, parsed, values):
+        params = super().buildSoapyDeviceParameters(parsed, values)
+
+        address, port = values["afedri_adress_port"].split(":")
```

Comment:
consider adding a default for the port. or maybe add a validator to the field that ensures that `:` is present. i expect this to result in ugly exceptions when the value isn't in the expected format.

Reply:
I've added proper validation.